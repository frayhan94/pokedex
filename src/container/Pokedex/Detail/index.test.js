import React from 'react';

import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import PokeDetail from './index';

const mockStore = configureMockStore();

describe('RENDER POKE DETAIL', () => {
  let wrapper;
  let store;

  beforeEach(() => {
    const initialState = {
      pokedexReducer: {
        loadingPokedexDetail: false,
        pokedexdetail: {
          base_experience: 4,
          weight: 10,
          height: 20,
          name: 'ABC',
          sprites: {
            front_default: 'http://google.com',
          },
        },
      },
    };
    store = mockStore(initialState);
    store.dispatch = jest.fn();
    wrapper = renderer.create(
      <Provider store={store}>
        <PokeDetail match={{ params: { id: 1 } }} />
      </Provider>,
    );
  });

  it('RENDER CORRECTLY', () => {
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
