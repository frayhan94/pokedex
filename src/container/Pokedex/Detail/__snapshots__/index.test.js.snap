// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RENDER POKE DETAIL RENDER CORRECTLY 1`] = `
<div
  className="Grid justifyCenter split--2"
>
  <div
    className="Card alignCenter justifyCenter"
  >
    <div
      className="cardMedia"
    >
      <img
        alt="ABC"
        src="http://google.com"
      />
    </div>
    <div
      className="Chat__Body"
    >
      <div
        className="List"
      >
        <div
          className="ListLeft"
        >
          Name
        </div>
        <div>
          ABC
        </div>
      </div>
      <div
        className="List"
      >
        <div
          className="ListLeft"
        >
          Experience
        </div>
        <div>
          4
        </div>
      </div>
      <div
        className="List"
      >
        <div
          className="ListLeft"
        >
          Weight
        </div>
        <div>
          10
        </div>
      </div>
      <div
        className="List"
      >
        <div
          className="ListLeft"
        >
          Height
        </div>
        <div>
          20
        </div>
      </div>
    </div>
  </div>
</div>
`;
